### PR TITLE
[machine] add integration test

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
- adds integration tests for  `eth-virtual-app-agreement-commitment`

### CODE REVIEW

File: packages/machine/test/integration/bignumber-jest-matcher.ts [hunk 1]
```diff
@@ -4,13 +4,21 @@ declare global {
   namespace jest {
     interface Matchers<R> {
       toBeEq(expected: BigNumber): BigNumber;
+      toBeBNAndEq(expected: BigNumber | number): BigNumber;
     }
   }
 }
 
+export function toBeBNAndEq(received: BigNumber | number, argument: BigNumber) {
```
Comment (snario) [inline packages/machine/test/integration/bignumber-jest-matcher.ts]: I think you could wrap `received` and `argument` in `bigNumberify` in `toBeEq` and get the same effect

File: packages/machine/test/integration/set-state.spec.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+expect.extend({ toBeEq });
+expect.extend({ toBeBNAndEq });
+
+beforeAll(async () => {
+  [{}, wallet, networkId] = await connectToGanache();
+
+  const relevantArtifacts = [
+    { contractName: "AppRegistry", ...AppRegistry },
+    { contractName: "ETHBucket", ...ETHBucket },
+    { contractName: "NonceRegistry", ...NonceRegistry },
```
Comment (snario) [inline packages/machine/test/integration/set-state.spec.ts]: I don't think the set state commitment needs this to be defined

File: packages/machine/test/integration/set-state.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  );
+});
+
+/**
+ * @summary Setup a StateChannel then set state on ETH Free Balance
+ */
+describe("Test", () => {
+  jest.setTimeout(JEST_TEST_WAIT_TIME);
+
+  it("test", async done => {
```
Comment (snario) [inline packages/machine/test/integration/set-state.spec.ts]: name

File: packages/machine/test/integration/set-state.spec.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    let freeBalanceETH = stateChannel.getFreeBalanceFor(AssetType.ETH);
+
+    const state = {
+      alice: stateChannel.multisigOwners[0],
+      bob: stateChannel.multisigOwners[1],
+      aliceBalance: WeiPerEther,
+      bobBalance: WeiPerEther
+    };
+
+    stateChannel = stateChannel.setState(freeBalanceETH.identityHash, state);
```
Comment (snario) [inline packages/machine/test/integration/set-state.spec.ts]: You can use my new helper fn `setFreeBalanceFor(AssetType, AppState)`

File: tsconfig.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
     "noImplicitAny": false,
-    "noUnusedLocals": true,
+    // "noUnusedLocals": true,
```
Comment (snario) [inline tsconfig.json]: If you can find a way to make this throw a warning but not fail I'm all for removing it

File: packages/machine/test/integration/set-state.spec.ts (hunk 4 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/machine/test/integration/set-state.spec.ts]: I wrote what I think is a better helper :p

File: packages/machine/src/ethereum/eth-virtual-app-agreement-commitment.ts [hunk 6]
```diff
@@ -42,13 +42,19 @@ export class ETHVirtualAppAgreementCommitment extends MultiSendCommitment {
...
+      throw Error("undefined terms");
+    }
     return {
       to: this.networkContext.ETHVirtualAppAgreement,
       value: 0,
       data: iface.functions.delegateTarget.encode([
         {
           registry: this.networkContext.AppRegistry,
-          // terms,
+          terms: this.terms!,
```
[the remark this answers is not in the document]
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/machine/src/ethereum/eth-virtual-app-agreement-commitment.ts]: fixed

File: packages/machine/src/ethereum/eth-virtual-app-agreement-commitment.ts [hunk 7]
```diff
@@ -54,7 +53,11 @@ export class ETHVirtualAppAgreementCommitment extends MultiSendCommitment {
       data: iface.functions.delegateTarget.encode([
         {
           registry: this.networkContext.AppRegistry,
-          terms: this.terms!,
+          terms: {
+              assetType: AssetType.ETH,
+              limit: new BigNumber(0),
```
Comment (cf19drofxots) [inline packages/machine/src/ethereum/eth-virtual-app-agreement-commitment.ts]: The limit is 0? Really?

Also, using `Zero` would remove needing to `new BigNumber`

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
Yes, read the comment in the solidity file

Le ven. 25 janv. 2019 1:59 a.m., cf19drofxots <notifications@github.com> a
écrit :

**Pull request merged**